### ISSUE TITLE
ASTIndex API does not fully provide for array types

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/annotation-tools#83]
```
ASTIndex is a map from AST nodes to ASTPaths.  Unfortunately, javac's 
representation of multi-dimensional arrays means that no tree node corresponds 
to inner array types.  For example, there is no node in the AST for int[][] 
that corresponds to int[].

The ASTPath format itself is capable of representing such inner types, and 
javac's implementations of ArrayTypeTree etc. include an alternate type 
representation with a nested structure.  Hence, one workaround from the API 
user's POV is to pass the top-level type and add as many instances of 
"ArrayType.type" as needed.  Since there is a workaround, and no one is 
currently blocked on this issue AFAIK, I am assigning low priority to this 
issue for now.
```

Original issue reported on code.google.com by `d...@cs.washington.edu` on 17 Apr 2014 at 10:41
